pytorch_lightning/core/lightning.py

```diff
@@ -1757,3 +1698,59 @@ def get_tqdm_dict(self) -> Dict[str, Union[int, str]]:
         rank_zero_warn("`get_tqdm_dict` was renamed to `get_progress_bar_dict` in v0.7.3"
                        " and this method will be removed in v1.0.0", DeprecationWarning)
         return self.get_progress_bar_dict()
+
+    # def auto_register_init_arguments(self, include_parents=False):
+    #     """Automatically register all init arguments to `self`."""
+    #     if not hasattr(self, '_module_self_arguments'):
+    #         self._auto_collect_arguments()
+    #
+    #     args = dict(self._module_parents_arguments) if include_parents else {}
+    #     args.update(self._module_self_arguments)
+    #     for k, v in ((k, v ) for k, v in args.items() if not hasattr(self, k)):
+    #         setattr(self, k, v)
+
+    def _auto_collect_arguments(self):
+        """Collect all arguments module arguments."""
+        frame = inspect.currentframe()
+
+        frame_args = _collect_init_args(frame, [])
+        child = _get_latest_child(frame)
+
+        # set module_arguments in child
+        child._module_self_arguments = frame_args[-1]
+        child._module_parents_arguments = {}
+        for args in frame_args[:-1]:
+            child._module_parents_arguments.update(args)
+
+    @property
+    def module_arguments(self) -> dict:
+        """Aggregate this module and all parents arguments."""
+        args = dict(self._module_parents_arguments)
+        args.update(self._module_self_arguments)
+        return args
+
+
+def _collect_init_args(frame, path_args: list) -> list:
+    """Recursive search for all children."""
+    if 'self' in frame.f_locals:
```

Comment:
maybe we should warn the user if we can't detect "self"

Reply:
this is for recursion and it is called from class so there is always one `self`